### DISCUSSION TITLE
チーム開発用のルール

### DISCUSSION THREAD

**Discussion by yoiyoicho** [nyohoo/NewBeLab-Vue#13] [General]
# Issueについて
 `enhancement` `question` `documentation`のタグを適切につける

# ブランチ名について

`[Issue番号]_[名前]_[内容]`
例）`01_fujiwara_environment`

# コミット名について
`[fix/add/update/remove]: [内容]`
例）`add: ログイン機能の追加`

fix：バグ修正
add：新規（ファイル）機能追加
update：機能修正（バグではない）
remove：削除（ファイル）

参考
https://qiita.com/itosho/items/9565c6ad2ffc24c09364
# プルリクについて

```
# Issue
close #[issue番号]

# やったこと
<!-- このプルリクエストにて何をしたのか？ -->

# 動作確認
<!-- どの環境でどんな動作チェックをしたか -->
<!-- スクショなどがあるとわかりやすくて良い -->

# その他
<!-- 関連するプルリクエストやイシュー、コンフルリンクなど、レビュワーがレビューするにあたっての補足情報 -->
```
参考
https://zenn.dev/stafes_blog/articles/d235146a37e0d5

# コンポーネント設計について
ディレクトリ構造
```
 L components/
   L parts/
   L templates/
   L views/
```

- **Parts**
parts ディレクトリに格納するのは「ロジックが無い UI コンポーネント」です。この構成における最小粒度のコンポーネントで他のコンポーネントには依存していません。スタイルを伴う汎用的なコンポーネントなので、ロジック等は内包しません。

- **Templates**
templates ディレクトリに格納するのは「コンポーネントを組み合わせることを目的としたコンポーネント」です。parts レイヤーと templates レイヤーに依存しており、部分的に切り出した範囲内のレイアウトを決めて、コンポーネント毎に依存するロジックもこちらで処理します。

- **Views**
views ディレクトリに格納するのは「テンプレートにデータを渡すことを目的としたコンポーネント」です。templates レイヤーに依存しており、API 通信などから取得したデータをそれぞれのテンプレートに割り振ることを責務にしています。

参考
https://zenn.dev/offers/articles/20220523-component-design-best-practice

**Comment by nyohoo** [comment]
プルリクのテンプレート文について、説明の部分はコメントアウトとなるよう変更しました
